esmf_regrid/experimental/io.py

```diff
@@ -0,0 +1,150 @@
+"""Provides load/save functions for regridders."""
+
+import iris
+from iris.coords import AuxCoord
+from iris.cube import Cube, CubeList
+from iris.experimental.ugrid import PARSE_UGRID_ON_LOAD
+import numpy as np
+import scipy.sparse
+
+from esmf_regrid.experimental.unstructured_scheme import (
+    GridToMeshESMFRegridder,
+    MeshToGridESMFRegridder,
+)
+
+
+def save_regridder(rg, filename):
+    """
+    Save a regridder scheme instance.
+
+    Saves either a `GridToMeshESMFRegridder` or a `MeshToGridESMFRegridder`.
+
+    Parameters
+    ----------
+    rg : GridToMeshESMFRegridder, MeshToGridESMFRegridder
+        The regridder instance to save.
+    filename : str
+        The file name to save to.
+    """
+    src_name = "regridder source field"
+    tgt_name = "regridder target field"
+    if isinstance(rg, GridToMeshESMFRegridder):
+        regridder_type = "grid to mesh"
```

Comment:
Could you not just use `rg.__class__.__name__` instead here?

That way it scales to supporting other regridders, and explicitly specific about the regridder under the hood.

Don't know if this matters... or whether you want the `regridder_type` just to be generic rather than specific to a class 🤔

Reply:
I'll change this for the time being because it tidies up some of the code, though this question does seem worth considering for when we are refining this.